mmrotate/datasets/dota.py

```diff
@@ -1,361 +1,167 @@
-                data_info['ann']['bboxes'] = []
-                data_info['ann']['labels'] = []
-                data_infos.append(data_info)
+                data_info['file_name'] = img_name
+                data_info['img_path'] = img_path
+                data_info['img_id'] = osp.split(img_path)[1]
+
+                img_bytes = self.file_client.get(img_path)
+                img = mmcv.imfrombytes(img_bytes, backend='cv2')
+                width, height = img.shape[:2]
+                del img, img_bytes
+                data_info['height'] = height
+                data_info['width'] = width
+
+                instances = []
+                instance = {}
+                instance['bbox'] = []
+                instance['bbox_label'] = []
+                instance['ignore_flag'] = 0
+                instances.append(instance)
+                data_info['instances'] = instances
+                data_list.append(data_info)
+
+            return data_list
         else:
-            for ann_file in ann_files:
+            txt_files = glob.glob(self.ann_file + '/*.txt')
+            if len(txt_files) == 0:
+                raise ValueError('There is no txt file in ' f'{self.ann_file}')
+            for txt_file in txt_files:
                 data_info = {}
-                img_id = osp.split(ann_file)[1][:-4]
+                img_id = osp.split(txt_file)[1][:-4]
+                data_info['img_id'] = img_id
                 img_name = img_id + '.png'
-                data_info['filename'] = img_name
-                data_info['ann'] = {}
-                gt_bboxes = []
-                gt_labels = []
-                gt_polygons = []
-                gt_bboxes_ignore = []
-                gt_labels_ignore = []
-                gt_polygons_ignore = []
-
-                if os.path.getsize(ann_file) == 0 and self.filter_empty_gt:
-                    continue
-
-                with open(ann_file) as f:
+                data_info['file_name'] = img_name
+                data_info['img_path'] = osp.join(self.data_prefix['img_path'],
+                                                 img_name)
+                img_bytes = self.file_client.get(data_info['img_path'])
+                img = mmcv.imfrombytes(img_bytes, backend='cv2')
+                width, height = img.shape[:2]
+                del img, img_bytes
+                data_info['height'] = height
+                data_info['width'] = width
+
+                instances = []
+                with open(txt_file) as f:
                     s = f.readlines()
                     for si in s:
+                        instance = {}
                         bbox_info = si.split()
-                        poly = np.array(bbox_info[:8], dtype=np.float32)
-                        try:
-                            x, y, w, h, a = poly2obb_np(poly, self.version)
-                        except:  # noqa: E722
-                            continue
+                        instance['bbox'] = np.array(
```

Comment:
The base dataset requires the data_list can be dumped to JSON. The box should be a list of float and converted to ndarray in the loading pipeline.